src/datasources/workspace/WorkspaceDataSource.ts

```diff
@@ -10,30 +12,38 @@ export class WorkspaceDataSource extends DataSourceBase<WorkspaceQuery> {
     readonly templateSrv: TemplateSrv = getTemplateSrv()
   ) {
     super(instanceSettings, backendSrv);
+
+    this.variables = new WorkspaceVariableSupport();
   }
 
   baseUrl = this.instanceSettings.url + '/niuser/v1';
 
-  defaultQuery = {
-    constant: 3.14,
-  };
-
-  async runQuery(query: WorkspaceQuery, { range }: DataQueryRequest): Promise<DataFrameDTO> {
-    return {
-      refId: query.refId,
-      fields: [
-        { name: 'Time', values: [range.from.valueOf(), range.to.valueOf()], type: FieldType.time },
-        { name: 'Value', values: [query.constant, query.constant], type: FieldType.number },
-      ],
-    };
+  defaultQuery = {};
+
+  async runQuery(_query: WorkspaceQuery, { app }: DataQueryRequest): Promise<DataFrame> {
+    return toDataFrame({ fields: this.workspaceToFields(app, await this.getWorkspaces()) });
   }
 
-  shouldRunQuery(query: WorkspaceQuery): boolean {
+  shouldRunQuery(): boolean {
     return true;
   }
 
   async testDatasource(): Promise<TestingStatus> {
     await this.get(this.baseUrl + '/workspaces');
     return { status: 'success', message: 'Data source connected and authentication successful!' };
   }
+
+  private workspaceToFields(app: string, workspaces: Workspace[]): Array<{ name: string, values: string[] }> {
+    const ids: string[] = [];
+    const names: string[] = [];
+    const queryVariableEditor = app === 'dashboard';
```

Comment:
I don't think this is sufficient, unfortunately. When you're in the panel editor, the value is `panel-editor`, but when you're viewing a dashboard, both the panel queries and the variable queries are `dashboard`.

For a variable query, the request looks like:
```
{
    "app": "dashboard",
    "requestId": "deb976da-36a0-42ea-bd05-8b4efb8ec870",
    "timezone": "",
    "range": {
        "from": "2023-09-27T13:49:13.572Z",
        "to": "2023-09-27T19:49:13.572Z",
        "raw": {
            "from": "now-6h",
            "to": "now"
        }
    },
    "interval": "",
    "intervalMs": 0,
    "targets": [
        {
            "refId": "variable-query"
        }
    ],
    "scopedVars": {
        "variable": {}
    },
    "startTime": 1695844153572
}
```
So we may be able to key off the `refId` property...but I'm not sure if we can rely on that not to change at some point in the future.